classes/services/NavigationMenuService.inc.php

```diff
@@ -64,13 +106,22 @@ function getDisplayStatusCallback($hookName, $args) {
 		$isUserLoggedIn = \Validation::isLoggedIn();
 		$isUserLoggedInAs = \Validation::isLoggedInAs();
 		$context = $request->getContext();
+		$contextId = $context ? $context->getId() : CONTEXT_ID_NONE;
 
 		$this->transformNavMenuItemTitle($templateMgr, $navigationMenuItem);
 
 		$menuItemType = $navigationMenuItem->getType();
 
 		if ($navigationMenuItem->getIsDisplayed()) {
 
+			$request = \Application::getRequest();
+			$context = $request->getContext();
+			$contextId = $context ? $context->getId() : CONTEXT_ID_NONE;
+
+			$menuItemType = $navigationMenuItem->getType();
+
+			$relatedObject = null;
```

Comment:
When testing, my NMIs for series and categories didn't show up. It doesn't look like `$relatedObject` is ever reset from `null`. Did you mean to remove this?